nullaway/src/main/java/com/uber/nullaway/handlers/InferredJARModelsHandler.java

```diff
@@ -91,13 +94,40 @@ public InferredJARModelsHandler() {
     }
   }
 
+  /*
+   * Scan Java Classpath for JarInfer model jars and map their names to locations
+   */
+  private void processClassPath(String[] args) {
+    Set<String> paths = new LinkedHashSet<>();
+    for (int i = 0; i < args.length; ++i)
+      if (args[i].equals("-classpath") || args[i].equals("-processorpath"))
+        paths.addAll(Arrays.asList(args[++i].split(":")));
+    for (String path : paths) {
+      String[] parts = path.split("/");
+      if (parts.length >= 2) {
+        String name = parts[parts.length - 2].replaceFirst("^__", "");
+        if (name.startsWith("jarinfer_model_")) {
+          name = name.replaceAll("(^jarinfer_model_|\\.[ja]ar(__|#class-abi)$)", "");
```

Comment:
This is probably fine for right now, but it hardcodes the specific type of paths used by buck and our set up, so long/medium term we need something more general to map code jar to astubx jar. Maybe a config flag? Or maybe we could just go ahead and load the models here and then search by class name? (we need to figure out what to do if we get conflicting models for the same class if it's found in two jars, probably just load the first one based on classpath rules and assume astubx jars appear in the same order as their code counterparts)

@kageiit @msridhar ?

Reply:
Lets use flags